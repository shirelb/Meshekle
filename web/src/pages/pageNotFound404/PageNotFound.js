import React from 'react';
import {Grid, Header, Image} from 'semantic-ui-react';

const PageNotFound = () => (
    <Grid stretched textAlign={"center"}>
        <Grid.Column stretched textAlign={"center"}>
            <Image centered src={require('../../images/logo1.png')} size={'large'}/>
            <Header size={"huge"} textAlign={"center"} icon style={{marginTop: "-10px"}}>
                משקל'ה
                <Header.Subheader>יש משק ויש משקל'ה</Header.Subheader>
            </Header>

            <div style={{marginTop: 20, textAlign: "center"}}>
                <Header as={'h1'}> 404 </Header>
                <Header as={'h2'}> Page not found </Header>
            </div>
        </Grid.Column>
    </Grid>
);

export default PageNotFound;