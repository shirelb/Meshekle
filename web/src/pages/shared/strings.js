const loginPageStrings = {
    LOGIN: "התחבר",
    SUBMIT: "שלח",
    WELCOME_TO_LOGIN: "Welcome to the login screen!",
    USER_ID_PLACEHOLDER: "ת.ז.",
    PASSWORD_PLACEHOLDER: "סיסמא",
    APP_NAME: "משקל\'ה",
    WRONG_CREDENTIALS: "אחד הפרטים שהזנת לא נכון",
    NO_SUCH_USER: "המשתמש אינו קיים",
    EMPTY_USER_ID: "הוסף ת.ז. ונסה להתחבר שוב",
    EMPTY_PASSWORD: "הוסף סיסמא ונסה להתחבר שוב",
};

const mainScreenStrings = {
    LOGOUT: "התנתק",
    WELCOME: "ברוכים הבאים ",
};

const drawerMenu = {
    LOGOUT: "התנתק",
    MainScreenName: "בית",
    AppointmentsScreenName: "תורים",
    ChoresScreenName: "תורנויות",
};

export default {loginPageStrings, mainScreenStrings, drawerMenu};