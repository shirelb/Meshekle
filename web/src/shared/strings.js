const loginPageStrings = {
    LOGIN: "התחבר",
    SUBMIT: "שלח",
    WELCOME_TO_LOGIN: "Welcome to the login screen!",
    USER_ID_PLACEHOLDER: "ת.ז.",
    PASSWORD_PLACEHOLDER: "סיסמא",
    APP_NAME: "משקל\'ה",
    WRONG_CREDENTIALS: "אחד הפרטים שהזנת לא נכון",
    NO_SUCH_USER: "המשתמש אינו קיים",
    EMPTY_USER_ID: "הוסף ת.ז. ונסה להתחבר שוב",
    EMPTY_PASSWORD: "הוסף סיסמא ונסה להתחבר שוב",
};

const mainPageStrings = {
    LOGOUT: "התנתק",
    WELCOME: "ברוכים הבאים ",
    MAIN_PAGE_TITLE: "בית",
    SETTINGS_PAGE_TITLE: "הגדרות",
    PHONE_BOOK_PAGE_TITLE: "ספר טלפונים",
    PHONE_BOOK_PAGE_USERS_TITLE: "משתמשים",
    PHONE_BOOK_PAGE_SERVICE_PROVIDERS_TITLE: "נותני שירות",
    APPOINTMENTS_PAGE_TITLE: "תורים",
    CHORES_PAGE_TITLE: "תורנויות",
    ANNOUNCEMENTS_PAGE_TITLE: "לוח מודעות",
    ROLES_HairDresser: "מספרה",
    ROLES_Admin: "מנהלה",
    ROLES_Dentist: "רפואת שיניים",
    ROLES_Undefined: "לא נמצאו תפקידים מתאימים",

};

const phoneBookPageStrings = {
    ADD_SERVICE_PROVIDER: "נותן שירות חדש",
    ADD_USER: "משתמש חדש",
    EDIT_USER: "ערוך משתמש",
    USER_ID_HEADER: "ת.ז.",
    FULLNAME_HEADER: "שם מלא",
    PASSWORD_HEADER: "סיסמא",
    EMAIL_HEADER: "אימייל",
    MAILBOX_HEADER: "תיבת דואר",
    CELLPHONE_HEADER: "פלאפון",
    PHONE_HEADER: "טלפון",
    BORN_DATE_HEADER: "תאריך לידה",
    ACTIVE_HEADER: "קיים",
    ACTIVE_ANSWER_YES: "כן",
    ACTIVE_ANSWER_NO: "לא",
    SERVICE_PROVIDER_ID_HEADER: "ת.ז. נותן שירות",
    SERVICE_PROVIDER_ROLE_HEADER: "תפקיד",
    SERVICE_PROVIDER_USER_ID_HEADER: "ת.ז. יוזר",
    SERVICE_PROVIDER_OPERATION_TIME_HEADER: "זמני פעילות",
    SERVICE_PROVIDER_APPOINTMENT_WAY_TYPE_HEADER: "דרך קביעת תורים",
};

const appointmentsPageStrings = {
    ADD_APPOINTMENT: "תור חדש",
    APPOINTMENT_ID: "מספר מזהה",
    CLIENT_NAME: "שם משתמש",
    CLIENT_ID: "ת.ז. משתמש",
    SERVICE_PROVIDER_ID: "ת.ז. נותן שירות",
    ROLE: "ענף",
    SUBJECT: "נושא",
    STATUS: "סטאטוס",
    DATE: "תאריך",
    START_TIME: "זמן התחלה",
    END_TIME: "זמן סיום",
    REMARKS: "הערות",
    OPTIONAL_TIMES: "זמנים אופציונאליים",
};

const announcementsPageStrings = {
    ANNOUNCE_REQ_TITLE_TABLE: "מודעות המחכות לאישור",
    ANNOUNCE_TITLE_TABLE: "מודעות באחריותך",
    ANNOUNCE_NUMBER:"מספר מודעה",
    ANNOUNCE_USERNAME:"שם המפרסם",
    ANNOUNCE_CATEGORY:"קטגוריה",
    ANNOUNCE_TITLE:"נושא",
    ANNOUNCE_CONTENT:"תוכן",
    ANNOUNCE_STATUS:"סטטוס",
    ANNOUNCE_FILE:"קובץ מצורף",
    ANNOUNCE_DATE_OF_EVENT:"תאריך אירוע",
    ANNOUNCE_EXPR_DATE:"תאריך תפוגה",
    OPERATION_OPTIONS:"אפשרויות",
    ADD_ANNOUNCEMENT:"פרסם מודעה חדשה",
};

const roles = {
    HairDresser: "מספרה",
    Admin: "מנהלה",
    Dentist: "רפואת שיניים",
};


const appointmentsWayType = {
    Dialog: "דיון",
    Slots: "שורה נעלמת",
    Fault: "פתיחת תקלה",
    Admin: "הכל",
};

export default {
    loginPageStrings,
    mainPageStrings,
    phoneBookPageStrings,
    appointmentsPageStrings,
    roles,
    appointmentsWayType,
    announcementsPageStrings,
};
