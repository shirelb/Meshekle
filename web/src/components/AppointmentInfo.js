import React from 'react';
import './styles.css';
import {Button, Modal} from 'semantic-ui-react';
import axios from 'axios';
import {Helmet} from 'react-helmet';
import {Table} from "semantic-ui-react/dist/commonjs/collections/Table";
import strings from "../shared/strings";
import getAppointmentByAppointmentID from "../shared/helpers";
import store from "store";


class UserInfo extends React.Component {
    constructor(props) {
        super(props);

        this.state = {event: this.props.location.state.event, appointment: this.props.location.state.appointment};

        this.handleDelete = this.handleDelete.bind(this);
        this.serviceProviderHeaders = {
            'Authorization': 'Bearer ' + store.get('serviceProviderToken')
        };
    }

    componentDidMount() {
        if (this.props.location.state.appointment)
            this.setState({appointment: this.props.location.state.appointment});
        else
            getAppointmentByAppointmentID(store.get('serviceProviderId'), this.props.match.params, this.serviceProviderHeaders)
                .then(({data: appointment}) => {
                    this.setState({appointment});
                });
    }

    handleDelete() {
        axios.delete('/api/users/1')
            .then(() => {
                console.log('user deleted');
            });
    }

    render() {
        const {event, appointment} = this.state;
        console.log('resder apponmnt info appointment ', appointment);

        return (
            <Modal open dimmer="blurring" closeIcon onClose={()=>this.props.history.goBack()}>
                <Helmet>
                    <title>Meshekle | {event.title}</title>
                </Helmet>

                <Modal.Header>{event.title}</Modal.Header>
                <Modal.Content image>
                    {/*<Image wrapped size="small" src={`https://api.adorable.io/avatars/250/${user.email}`}/>*/}
                    <Modal.Description>
                        <p>{strings.appointmentsPageStrings.APPOINTMENT_ID}: {appointment.appointmentId}</p>
                        <p>{strings.appointmentsPageStrings.CLIENT_NAME}: {event.title}</p>
                        <p>{strings.appointmentsPageStrings.SERVICE_PROVIDER_ID}: {appointment.AppointmentDetail.serviceProviderId}</p>
                        <p>{strings.appointmentsPageStrings.ROLE}: {appointment.AppointmentDetail.role}</p>
                        <p>{strings.appointmentsPageStrings.SUBJECT}: {appointment.AppointmentDetail.subject}</p>
                        <p>{strings.appointmentsPageStrings.STATUS}: {appointment.status}</p>
                        <p>{strings.appointmentsPageStrings.DATE}: {new Date(appointment.startDateAndTime).toISOString().split('T')[0]}                         </p>
                        <p>{strings.appointmentsPageStrings.START_TIME}: {new Date(appointment.startDateAndTime).toISOString().split('T')[1].split('.')[0].slice(0, -3)}                         </p>
                        <p>{strings.appointmentsPageStrings.END_TIME}: {new Date(appointment.endDateAndTime).toISOString().split('T')[1].split('.')[0].slice(0, -3)}                         </p>
                        <p>{strings.appointmentsPageStrings.REMARKS}: {appointment.remarks}                         </p>
                    </Modal.Description>
                </Modal.Content>
                <Modal.Actions className='alignLeft'>
                    {/*<Button positive>Edit</Button>*/}
                    <Button negative onClick={this.handleDelete}>Delete</Button>
                    <Button positive onClick={()=>this.props.history.goBack()}>OK</Button>
                </Modal.Actions>
            </Modal>
        );
    }
}

export default UserInfo;