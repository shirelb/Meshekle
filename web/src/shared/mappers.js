import strings from "./strings";

var loginPageMapper = function (value) {
    switch (value) {
        case "userId doesn't exist!":
            return strings.loginPageStrings.NO_SUCH_USER;
        default:
            return strings.loginPageStrings.WRONG_CREDENTIALS;
    }
};

var mainPageMapper = function (value) {
    switch (value) {
        default:
            return value;
    }
};

var daysMapper = function (value) {
    switch (value) {
        case "Sunday":
            return strings.days.Sunday;
        case "Monday":
            return strings.days.Monday;
        case "Tuesday":
            return strings.days.Tuesday;
        case "Wednesday":
            return strings.days.Wednesday;
        case "Thursday":
            return strings.days.Thursday;
        case "Friday":
            return strings.days.Friday;
        case "Saturday":
            return strings.days.Saturday;
        default:
            return strings.days.Undefined;
    }
};

var appointmentStatusMapper = function (value) {
    switch (value) {
        case "set":
            return "נקבע";
        case "cancelled":
            return "בוטל";
        default:
            return value;
    }
};

var appointmentRequestStatusMapper = function (value) {
    switch (value) {
        case "requested":
            return "בוקש";
        case "approved":
            return "אושר";
        case "rejected":
            return "נדחה";
        default:
            return value;
    }
};

export default {loginPageMapper, mainPageMapper, daysMapper, appointmentStatusMapper, appointmentRequestStatusMapper};
