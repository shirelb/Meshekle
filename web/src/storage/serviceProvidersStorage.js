import axios from "axios";
import {SERVER_URL} from "../shared/constants";
import store from "store";

const serviceProviderHeaders = {
    'Authorization': 'Bearer ' + store.get('serviceProviderToken')
};

var getServiceProviders = function () {
    return axios.get(`${SERVER_URL}/api/serviceProviders`,
        {headers: serviceProviderHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get serviceProviders error ', error)
        });
};

var getRolesOfServiceProvider = (serviceProviderId) => {
    return axios.get(`${SERVER_URL}/api/serviceProviders/roles/serviceProviderId/${serviceProviderId}`,
        {headers: serviceProviderHeaders}
    )
        .then((response) => {
            let roles = response.data;
            console.log('roles ', roles);
            return roles;
        })
        .catch((error) => {
            console.log('get serviceProvider roles error ', error)
        });
};

var serviceProviderLogin = (userId, password) => {
    return axios.post(`${SERVER_URL}/api/serviceProviders/login/authenticate`,
        {
            "userId": userId,
            "password": password
        },
    );
};

var serviceProviderValidToken = function (token) {
    return axios.post(`${SERVER_URL}/api/serviceProviders/validToken`,
        {
            "token": token
        },
    );
}

var getServiceProviderPermissionsById = (serviceProviderId) => {
    return axios.get(`${SERVER_URL}/api/serviceProviders/serviceProviderId/${serviceProviderId}/permissions`,
        {headers: serviceProviderHeaders}
    )
        .then((response) => {
            let permissions = response.data;
            console.log('permissions ', permissions);
            return permissions;
        })
        .catch((error) => {
            console.log('error ', error);
        });
};


export default {
    getServiceProviders,
    getRolesOfServiceProvider,
    serviceProviderLogin,
    serviceProviderValidToken,
    getServiceProviderPermissionsById,

};
