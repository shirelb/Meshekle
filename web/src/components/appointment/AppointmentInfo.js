import React from 'react';
import '../styles.css';
import {Button, Container, Message, Modal} from 'semantic-ui-react';
import {Helmet} from 'react-helmet';
import strings from "../../shared/strings";
import store from "store";
import AppointmentEdit from "./AppointmentEdit";
import appointmentsStorage from "../../storage/appointmentsStorage";
import moment from 'moment';
import {Route, Switch} from "react-router-dom";
import mappers from "../../shared/mappers";
import usersStorage from "../../storage/usersStorage";


class AppointmentInfo extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            // appointment: this.props.location.state ? this.props.location.state.appointment : {},
            infoError: false,
            infoErrorHeader: '',
            infoErrorContent: ''
        };

        this.handleDelete = this.handleDelete.bind(this);
        this.handleEdit = this.handleEdit.bind(this);

        this.serviceProviderHeaders = {
            'Authorization': 'Bearer ' + store.get('serviceProviderToken')
        };
    }

    componentDidMount() {
        if (this.props.location.state) {
            if (this.props.location.state.appointment)
                this.setState({appointment: this.props.location.state.appointment});
        } else
            appointmentsStorage.getAppointmentByAppointmentID(store.get('serviceProviderId'), this.props.match.params.appointmentId, this.serviceProviderHeaders)
                .then((appointment) => {
                    if (appointment.response) {
                        if (appointment.response.status !== 200)
                            this.setState({
                                infoError: true,
                                infoErrorHeader: 'קרתה שגיאה בעת הבאת פרטי התור',
                                infoErrorContent: mappers.errorMapper(appointment.response)
                            });
                    } else {
                        usersStorage.getUserByUserID(appointment.AppointmentDetail.clientId, this.serviceProviderHeaders)
                            .then(user => {
                                if (user.response) {
                                    if (user.response.status !== 200)
                                        this.setState({
                                            infoError: true,
                                            infoErrorHeader: 'קרתה שגיאה בעת הבאת פרטי התור',
                                            infoErrorContent: mappers.errorMapper(user.response)
                                        });
                                } else {
                                    appointment.clientName = user.fullname;
                                    this.setState({appointment: appointment});
                                }
                            });
                    }
                });
    }

    handleDelete() {
        // console.log('appointment handleDelete ', this.serviceProviderHeaders);
        appointmentsStorage.cancelAppointmentById(this.state.appointment, this.serviceProviderHeaders)
            .then((response) => {
                // console.log('appointment handleDelete ', response.data);
                if (response.response) {
                    if (response.response.status !== 200)
                        this.setState({
                            infoError: true,
                            infoErrorHeader: 'קרתה שגיאה בעת מחיקת התור',
                            infoErrorContent: mappers.errorMapper(response.response)
                        });
                } else
                    this.props.history.goBack();
            });
    }

    handleEdit() {
        this.props.history.push(`${this.props.match.url}/edit`, {
            appointment: this.state.appointment,
            openedFrom: "AppointmentInfo"
        });
    }

    render() {
        const {appointment, infoError, infoErrorHeader, infoErrorContent} = this.state;

        return (
            <div>
                <Modal open dimmer="blurring" closeIcon onClose={() => this.props.history.goBack()}>
                    <Helmet>
                        <title>Meshekle | פרטי תור {appointment ? appointment.clientName ? appointment.clientName : appointment.AppointmentDetail.clientId : ""}</title>
                    </Helmet>

                    <Modal.Header>{appointment ? appointment.clientName ? appointment.clientName : appointment.AppointmentDetail.clientId : ""}</Modal.Header>
                    <Modal.Content image>
                        {/*<Image wrapped size="small" src={`https://api.adorable.io/avatars/250/${user.email}`}/>*/}
                        {appointment ?
                            <Modal.Description>
                                <Container text>
                                    <p>{strings.appointmentsPageStrings.APPOINTMENT_ID}: {appointment.appointmentId}</p>
                                    <p>{strings.appointmentsPageStrings.CLIENT_NAME}: {appointment.clientName}</p>
                                    <p>{strings.appointmentsPageStrings.SERVICE_PROVIDER_ID}: {appointment.AppointmentDetail.serviceProviderId}</p>
                                    <p>{strings.appointmentsPageStrings.ROLE}: {strings.roles[appointment.AppointmentDetail.role]}</p>
                                    <p>{strings.appointmentsPageStrings.SUBJECT}: {JSON.parse(appointment.AppointmentDetail.subject).join(", ")}</p>
                                    <p>{strings.appointmentsPageStrings.STATUS}: {mappers.appointmentStatusMapper(appointment.status)}</p>
                                    <p>{strings.appointmentsPageStrings.DATE}: {moment(appointment.startDateAndTime).format('DD.MM.YYYY')}</p>
                                    <p>{strings.appointmentsPageStrings.START_TIME}: {moment(appointment.startDateAndTime).format("HH:mm")} </p>
                                    <p>{strings.appointmentsPageStrings.END_TIME}: {moment(appointment.endDateAndTime).format("HH:mm")}</p>
                                    <p>{strings.appointmentsPageStrings.REMARKS}: {appointment.remarks}</p>

                                    {infoError ?
                                        <Message
                                            error
                                            header={infoErrorHeader}
                                            content={infoErrorContent}
                                        />
                                        : null
                                    }
                                </Container>
                            </Modal.Description> :
                            <Modal.Description/>
                        }
                    </Modal.Content>
                    <Modal.Actions className='alignLeft'>
                        <Button positive onClick={this.handleEdit}>ערוך</Button>
                        <Button negative onClick={this.handleDelete}>מחק</Button>
                        {/*<Button positive onClick={() => this.props.history.goBack()}>השאר</Button>*/}
                    </Modal.Actions>
                </Modal>

                <div>
                    <Switch>
                        <Route exec path={`${this.props.match.url}/edit`}
                               render={(props) => (
                                   <AppointmentEdit
                                       {...props}
                                       userOptions={this.props.userOptions}
                                       getUsersForAppointmentForm={this.props.getUsersForAppointmentForm}
                                       serviceProviderRoles={this.props.serviceProviderRoles}
                                       getServiceProviderRoles={this.props.getServiceProviderRoles}
                                   />
                               )}/>
                        {/*<Redirect to={`${this.props.match.path}`}/>*/}
                    </Switch>
                </div>
            </div>
        );
    }
}

export default AppointmentInfo;