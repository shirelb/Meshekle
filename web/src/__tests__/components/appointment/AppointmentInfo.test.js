import React from 'react';
import {shallow} from "enzyme/build";
import {setupComponent} from "../../testHelpers";
import PageNotFound from "../../../pages/pageNotFound404/PageNotFound";
import store from 'store';
import moment from 'moment';
import {createMemoryHistory} from 'history';
import AppointmentInfo from "../../../components/appointment/AppointmentInfo";
import serviceProvidersStorage from "../../../storage/serviceProvidersStorage";
import appointmentsStorage from "../../../storage/appointmentsStorage";
import usersStorage from "../../../storage/usersStorage";
import users from "../../jsons/users";
import serviceProviders from "../../jsons/serviceProviders";
import appointmentsOf549963652 from "../../jsons/appointmentsServiceProvider549963652";
import strings from "../../../shared/strings";
import mappers from "../../../shared/mappers";

const appointmentRequestsOf549963652 = require("../../jsons/appointmentRequestsServiceProvider549963652");

jest.mock("store");
jest.mock("../../../storage/usersStorage");
jest.mock("../../../storage/serviceProvidersStorage");
jest.mock("../../../storage/appointmentsStorage");


describe("AppointmentInfo should", () => {
    let wrapper = null;
    let componentInstance = null;
    let appointmentTest = appointmentsOf549963652[0];
    appointmentTest.clientName = "Padget Creaser";
    const history = createMemoryHistory({
        initialEntries: ['/', '/home', '/appointments', '/appointments/428'],
        initialIndex: 3,
        keyLength: 10,
        getUserConfirmation: null
    });
    const props = {
        history: history,
        location: {
            pathname: '/appointments/428',
            state: {
                appointment: appointmentTest,
            }
        },
        match: {
            isExact: true,
            path: '/appointments/428',
            url: '/appointments/428',
        },
        serviceProviderRoles: ["appointmentsHairDresser"],
    };
    const mockStore = {
        serviceProviderId: "549963652",
        userId: "549963652",
    };

    store.get.mockImplementation((key) => mockStore[key]);
    usersStorage.getUsers.mockResolvedValue(users);
    usersStorage.getUserByUserID.mockImplementation((userId) => Promise.resolve(users.filter(user => user.userId === userId)[0]));
    serviceProvidersStorage.getServiceProviderUserDetails.mockImplementation((serviceProviderId) => Promise.resolve({data: users.filter(user => user.userId === serviceProviderId)[0]}));
    serviceProvidersStorage.getServiceProviderById.mockImplementation((serviceProviderId) => Promise.resolve(serviceProviders.filter(provider =>
        provider.serviceProviderId === serviceProviderId)));
    serviceProvidersStorage.getRolesOfServiceProvider.mockResolvedValue(["appointmentsHairDresser"]);
    appointmentsStorage.setAppointment.mockResolvedValue({});
    appointmentsStorage.getServiceProviderAppointments.mockResolvedValue({data: appointmentsOf549963652});
    appointmentsStorage.getServiceProviderAppointmentRequests.mockImplementation(() => Promise.resolve({data: JSON.parse(JSON.stringify(appointmentRequestsOf549963652))}));
    appointmentsStorage.updateAppointment.mockResolvedValue({});
    appointmentsStorage.getAppointmentByAppointmentID.mockResolvedValue({data: appointmentsOf549963652[0]});
    appointmentsStorage.cancelAppointmentById.mockResolvedValue({});

    beforeAll((done) => {
        done();
    });

    afterAll(() => {
    });

    beforeEach((done) => {
        jest.clearAllMocks();
        done();
    });

    afterEach(() => {
    });

    test('match snapshot', async () => {
        const props = {
            location: {
                pathname: '/appointments/428',
                state: {
                    appointment: appointmentTest
                }
            },
            match: {
                isExact: true,
                path: '/appointments/428',
                url: '/appointments/428',
            },
            serviceProviderRoles: ["appointmentsHairDresser"],
        };

        const arrResponse = setupComponent('shallow', AppointmentInfo, null, props, "/appointments/428");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        expect(componentInstance).toMatchSnapshot();
    });

    test("renders AppointmentInfo for /appointments/428", async () => {
        const arrResponse = await setupComponent('shallow', AppointmentInfo, history, props, "/appointments/428");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        expect(wrapper.find(AppointmentInfo)).toHaveLength(1);
        expect(wrapper.find(PageNotFound)).toHaveLength(0);
    });

    test("mounted with the right data", async () => {
        const arrResponse = await setupComponent('shallow', AppointmentInfo, history, props, "/appointments/428");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        expect(componentInstance.state.appointment).toBeDefined();
        expect(componentInstance.state.appointment.appointmentId).toEqual(appointmentTest.appointmentId);
        expect(componentInstance.state.appointment.clientName).toEqual(appointmentTest.clientName);
        expect(componentInstance.state.appointment.endDateAndTime).toEqual(appointmentTest.endDateAndTime);
        expect(componentInstance.state.appointment.remarks).toEqual(appointmentTest.remarks);
        expect(componentInstance.state.appointment.startDateAndTime).toEqual(appointmentTest.startDateAndTime);
        expect(componentInstance.state.appointment.status).toEqual(appointmentTest.status);
        expect(componentInstance.state.appointment.AppointmentDetail).toBeDefined();
        expect(componentInstance.state.appointment.AppointmentDetail.clientId).toEqual(appointmentTest.AppointmentDetail.clientId);
        expect(componentInstance.state.appointment.AppointmentDetail.role).toEqual(appointmentTest.AppointmentDetail.role);
        expect(componentInstance.state.appointment.AppointmentDetail.serviceProviderId).toEqual(appointmentTest.AppointmentDetail.serviceProviderId);
        expect(componentInstance.state.appointment.AppointmentDetail.subject).toEqual(appointmentTest.AppointmentDetail.subject);
    });

    test("mounted with the right data, without appointment in props", async () => {
        appointmentsStorage.getAppointmentByAppointmentID.mockResolvedValue(appointmentsOf549963652[0]);

        const props = {
            history: history,
            location: {
                pathname: '/appointments/428',
                state: undefined
            },
            match: {
                isExact: true,
                path: '/appointments/428',
                url: '/appointments/428',
                params:{
                    appointmentId:428
                }
            },
            serviceProviderRoles: ["appointmentsHairDresser"],
        };
        const arrResponse = await setupComponent('shallow', AppointmentInfo, history, props, "/appointments/428");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        expect(componentInstance.state.appointment).toBeDefined();
        expect(componentInstance.state.appointment.appointmentId).toEqual(appointmentTest.appointmentId);
        expect(componentInstance.state.appointment.clientName).toEqual(appointmentTest.clientName);
        expect(componentInstance.state.appointment.endDateAndTime).toEqual(appointmentTest.endDateAndTime);
        expect(componentInstance.state.appointment.remarks).toEqual(appointmentTest.remarks);
        expect(componentInstance.state.appointment.startDateAndTime).toEqual(appointmentTest.startDateAndTime);
        expect(componentInstance.state.appointment.status).toEqual(appointmentTest.status);
        expect(componentInstance.state.appointment.AppointmentDetail).toBeDefined();
        expect(componentInstance.state.appointment.AppointmentDetail.clientId).toEqual(appointmentTest.AppointmentDetail.clientId);
        expect(componentInstance.state.appointment.AppointmentDetail.role).toEqual(appointmentTest.AppointmentDetail.role);
        expect(componentInstance.state.appointment.AppointmentDetail.serviceProviderId).toEqual(appointmentTest.AppointmentDetail.serviceProviderId);
        expect(componentInstance.state.appointment.AppointmentDetail.subject).toEqual(appointmentTest.AppointmentDetail.subject);
    });

    test("handleDelete", async () => {
        const arrResponse = await setupComponent('shallow', AppointmentInfo, history, props, "/appointments/428");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        await componentInstance.handleDelete();
        expect(componentInstance.props.history.location.pathname).toEqual('/appointments');
    });

    test("handleEdit", async () => {
        const arrResponse = await setupComponent('shallow', AppointmentInfo, history, props, "/appointments/428");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        await componentInstance.handleEdit();
        expect(componentInstance.props.history.location.pathname).toEqual('/appointments/428/edit');
    });

    test("handleDelete on click delete button", async () => {
        const arrResponse = await setupComponent('shallow', AppointmentInfo, history, props, "/appointments/428");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        await wrapper.find('AppointmentInfo').dive().find('Button').at(1).simulate('click')
        expect(componentInstance.props.history.location.pathname).toEqual('/appointments');
    });

    test("handleEdit on click edit button", async () => {
        const arrResponse = await setupComponent('shallow', AppointmentInfo, history, props, "/appointments/428");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        await wrapper.find('AppointmentInfo').dive().find('Button').at(0).simulate('click')
        expect(componentInstance.props.history.location.pathname).toEqual('/appointments/428/edit');
    });

    test("render with what the user see", async () => {
        const arrResponse = await setupComponent('shallow', AppointmentInfo, history, props, "/appointments/428");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        const shallowWrapper = wrapper.find('AppointmentInfo').dive();

        expect(shallowWrapper.find('Modal')).toHaveLength(1);
        expect(shallowWrapper.find('ModalHeader')).toHaveLength(1);
        expect(shallowWrapper.find('ModalHeader').props().children).toEqual(appointmentTest.clientName);
        expect(shallowWrapper.find('ModalContent')).toHaveLength(1);
        expect(shallowWrapper.find('p')).toHaveLength(10);
        expect(shallowWrapper.find('p').at(0).props().children[0]).toEqual(strings.appointmentsPageStrings.APPOINTMENT_ID);
        expect(shallowWrapper.find('p').at(0).props().children[2]).toEqual(appointmentTest.appointmentId);
        expect(shallowWrapper.find('p').at(1).props().children[0]).toEqual(strings.appointmentsPageStrings.CLIENT_NAME);
        expect(shallowWrapper.find('p').at(1).props().children[2]).toEqual(appointmentTest.clientName);
        expect(shallowWrapper.find('p').at(2).props().children[0]).toEqual(strings.appointmentsPageStrings.SERVICE_PROVIDER_ID);
        expect(shallowWrapper.find('p').at(2).props().children[2]).toEqual(appointmentTest.AppointmentDetail.serviceProviderId);
        expect(shallowWrapper.find('p').at(3).props().children[0]).toEqual(strings.appointmentsPageStrings.ROLE);
        expect(shallowWrapper.find('p').at(3).props().children[2]).toEqual(strings.roles[appointmentTest.AppointmentDetail.role]);
        expect(shallowWrapper.find('p').at(4).props().children[0]).toEqual(strings.appointmentsPageStrings.SUBJECT);
        expect(shallowWrapper.find('p').at(4).props().children[2]).toEqual(JSON.parse(appointmentTest.AppointmentDetail.subject).join(", "));
        expect(shallowWrapper.find('p').at(5).props().children[0]).toEqual(strings.appointmentsPageStrings.STATUS);
        expect(shallowWrapper.find('p').at(5).props().children[2]).toEqual(mappers.appointmentStatusMapper(appointmentTest.status));
        expect(shallowWrapper.find('p').at(6).props().children[0]).toEqual(strings.appointmentsPageStrings.DATE);
        expect(shallowWrapper.find('p').at(6).props().children[2]).toEqual(moment(appointmentTest.startDateAndTime).format('DD.MM.YYYY'));
        expect(shallowWrapper.find('p').at(7).props().children[0]).toEqual(strings.appointmentsPageStrings.START_TIME);
        expect(shallowWrapper.find('p').at(7).props().children[2]).toEqual(moment(appointmentTest.startDateAndTime).format("HH:mm"));
        expect(shallowWrapper.find('p').at(8).props().children[0]).toEqual(strings.appointmentsPageStrings.END_TIME);
        expect(shallowWrapper.find('p').at(8).props().children[2]).toEqual(moment(appointmentTest.endDateAndTime).format("HH:mm"));
        expect(shallowWrapper.find('p').at(9).props().children[0]).toEqual(strings.appointmentsPageStrings.REMARKS);
        expect(shallowWrapper.find('p').at(9).props().children[2]).toEqual(appointmentTest.remarks);
        expect(shallowWrapper.find('ModalActions')).toHaveLength(1);
        expect(shallowWrapper.find('Button')).toHaveLength(2);
        expect(shallowWrapper.find('Button').at(0).props().children).toEqual("ערוך");
        expect(shallowWrapper.find('Button').at(0).props().positive).toBeTruthy();
        expect(shallowWrapper.find('Button').at(1).props().children).toEqual("מחק");
        expect(shallowWrapper.find('Button').at(1).props().negative).toBeTruthy();
    });
});
