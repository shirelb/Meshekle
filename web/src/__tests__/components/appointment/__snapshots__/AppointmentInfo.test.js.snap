// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentInfo should match snapshot 1`] = `undefined`;
